### PR TITLE
added OneMatrix for matrix expressions with only ones

### PR DESCRIPTION
**Pull request by Upabjojr**
#### Release Notes

* matrices
  * Introduction of `OneMatrix`, a matrix symbol representing matrices of only 1 entries.

### CODE REVIEW

File: sympy/matrices/expressions/tests/test_matexpr.py [hunk 1]
```diff
@@ -76,6 +76,40 @@ def test_ZeroMatrix_doit():
...
+def test_OneMatrix():
+    A = MatrixSymbol('A', n, m)
+    a = MatrixSymbol('a', n, 1)
+    U = OneMatrix(n, m)
+
+    assert U.shape == (n, m)
+
+    assert isinstance(A + U, Add)
+    # assert a*OneMatrix(n, 1) == a
+    # assert OneMatrix(n, 1)*a == a
```
Comment (sylee957) [inline sympy/matrices/expressions/tests/test_matexpr.py]: Would these commented-out tests should be made more meaningful?

And I don't think that the matrix multiplication can be defined here, unless n is 1.

File: sympy/matrices/expressions/tests/test_matexpr.py [hunk 2]
```diff
@@ -76,6 +76,40 @@ def test_ZeroMatrix_doit():
...
+    assert U.conjugate() == U
+
+    assert OneMatrix(n, n) ** 0 == Identity(n)
+    with raises(ShapeError):
+        U ** 0
+    with raises(ShapeError):
+        U ** 2
+
+    U = OneMatrix(n, n)
+    assert U[1, 2] == 1
```
Comment (sylee957) [inline sympy/matrices/expressions/tests/test_matexpr.py]: I am not sure that matrices with symbolic shape can have its elements defined at explicit positions.

**Review by sylee957 — COMMENTED** [review]
Looks good.
But I think you may need to add to `__init__.py`, so that it can be loaded under sympy main module.

File: sympy/matrices/expressions/tests/test_matexpr.py (hunk 1 above)
Comment (Upabjojr) [reply sympy/matrices/expressions/tests/test_matexpr.py]: These are wrong. I shall remove them.

File: sympy/matrices/expressions/tests/test_matexpr.py (hunk 2 above)
Comment (Upabjojr) [reply sympy/matrices/expressions/tests/test_matexpr.py]: `ZeroMatrix` behaves like this (always returns zero). There should be a bounds check, but I don't think this is important. If you access a matrix containing only one element, you're pretty sure you get that element.

**Pull request merged**